plugins/inputs/t128_metrics/t128_metrics_test.go

```diff
@@ -0,0 +1,512 @@
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, "{}", `[{
+			"id": "/stats/test-metric",
+			"permutations": [{
+				"parameters": [],
+				"value": "50"
+			}]
+		}]`}},
+		ExpectedMetrics: []*testutil.Metric{
+			{
+				Measurement: "test-metric",
+				Tags:        map[string]string{},
+				Fields:      map[string]interface{}{"test-field": 50},
+			},
+		},
+		ExpectedErrors: nil,
+	},
+	{
+		Name: "forms float value if it is numeric",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, "{}", `[{
+			"id": "/stats/test-metric",
+			"permutations": [{
+				"parameters": [],
+				"value": "50.5"
+			}]
+		}]`}},
+		ExpectedMetrics: []*testutil.Metric{
+			{
+				Measurement: "test-metric",
+				Tags:        map[string]string{},
+				Fields:      map[string]interface{}{"test-field": 50.5},
+			},
+		},
+		ExpectedErrors: nil,
+	},
+	{
+		Name: "adds permutation parameters to metrics",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, "{}", `[{
+			"id": "/stats/test-metric",
+			"permutations": [{
+				"parameters": [
+					{
+						"name": "node",
+						"value": "node1"
+					},
+					{
+						"name": "interface",
+						"value": "intf1"
+					}
+				],
+				"value": "0"
+			}]
+		}]`}},
+		ExpectedMetrics: []*testutil.Metric{
+			{
+				Measurement: "test-metric",
+				Tags:        map[string]string{"node": "node1", "interface": "intf1"},
+				Fields:      map[string]interface{}{"test-field": 0},
+			},
+		},
+		ExpectedErrors: nil,
+	},
+	{
+		Name: "produces multiple metrics for multiple permutations",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, "{}", `[{
+			"id": "/stats/test",
+			"permutations": [
+				{
+					"parameters": [
+						{
+							"name": "node",
+							"value": "node1"
+						}
+					],
+					"value": "897"
+				},
+				{
+					"parameters": [
+						{
+							"name": "node",
+							"value": "node2"
+						}
+					],
+					"value": "306"
+				}
+			]
+		}]`}},
+		ExpectedMetrics: []*testutil.Metric{
+			{
+				Measurement: "test-metric",
+				Tags:        map[string]string{"node": "node1"},
+				Fields:      map[string]interface{}{"test-field": 897},
+			},
+			{
+				Measurement: "test-metric",
+				Tags:        map[string]string{"node": "node2"},
+				Fields:      map[string]interface{}{"test-field": 306},
+			},
+		},
+		ExpectedErrors: nil,
+	},
+	{
+		Name: "hits multiple endpoints",
```

Comment:
is there a test where the plugin hits multiple endpoints but one of them replies with error or empty etc?

Reply:
nevermind - I see `mixes errors and valid results`